docs/source/scilla-in-depth.rst

```diff
@@ -120,7 +120,7 @@ variables are to be specified when the contract is deployed.
 .. note::
 
    In addition to the explicitly declared immutable fields, a Scilla
-   contract has an implicitly declared mutable field ``_this_address``
+   contract has an implicitly declared immutable field ``_this_address``
```

Comment:
Good catch! Actually `_this_address` is a contract parameter and not a field -- to access it one does not need to dereference it using `<-` primitive.

Reply:
Is it then more accurate to say "...immutable contract parameter ``_this_address``..."?

Reply:
Correct! It may require changing some text below, though.

Reply:
What's the diff between contract parameter and contract variables? It is mentioned [here](https://scilla.readthedocs.io/en/latest/scilla-by-example.html#adding-another-transition) that to read from the contract state, you need use`<-`

Reply:
"Contract state variable" means "mutable field" here, so that's part of a contract's state that can be modified via `:=` or `accept` statement in case of the implicit mutable field `_balance`.

Contract parameters are basically constants and cannot be modified, so neither `<-` not `:=` make sense for these.

The terminology is a bit inconsistent here. Thank for pointing that out.